concourse/steps/scan_sources.py

```diff
@@ -63,98 +66,156 @@ def scan_component_with_whitesource(
+            file=tmp_file,
+            filename=component_filename,
             product_token=product_token,
-            component=component,
+            project_name=scan_artifact.name,
+            requester_email=requester_mail,
         )
+        # TODO save scanned commit hash or tag in project tag show scanned version
+        # version for agent will create a new project
+        # https://whitesource.atlassian.net/wiki/spaces/WD/pages/34046170/HTTP+API+v1.1#HTTPAPIv1.1-ProjectTags
+
+
+def notify_users(
+    ws_client: whitesource.client.WhitesourceClient,
+    product_token: str,
+    cve_threshold: float,
+    notification_recipients: typing.List[str],
+    product_name: str
+):
+    # get all projects
+    ci.util.info('retrieving all projects')
+    projects = ws_client.get_all_projects_of_product(product_token=product_token)
+
+    # generate reporting table for console
+    tables = whitesource.util.generate_reporting_tables(
+        projects=projects,
+        threshold=cve_threshold,
+        tablefmt='simple',
+    )
 
-        # store in tmp file
-        with tempfile.TemporaryFile() as tmp_file:
-
-            ci.util.info('guessing commit hash')
-            # guess git-ref for the given component's version
-            commit_hash = product.util.guess_commit_from_ref(component)
-
-            # download whitesource component
-            ci.util.info('downloading component for scan')
-            whitesource.component.download_component(
-                github_api=post_project_object.github_api,
-                github_repo=post_project_object.github_repo,
-                dest=tmp_file,
-                ref=commit_hash,
-            )
-
-            # POST component>
-            ci.util.info('POST project')
-            post_project_object.whitesource_client.post_product(
-                product_token=post_project_object.product_token,
-                component_name=component.name,
-                component_version=post_project_object.component_version,
-                requester_email=requester_mail,
-                extra_whitesource_config=extra_whitesource_config,
-                file=tmp_file,
-            )
-
-        # get all projects
-        ci.util.info('retrieving all projects')
-        projects = client.get_all_projects(product_token=product_token)
-
-        # generate reporting table for console
+    whitesource.util.print_cve_tables(tables=tables)
+
+    if len(notification_recipients) > 0:
+        # generate reporting table for notification
         tables = whitesource.util.generate_reporting_tables(
             projects=projects,
             threshold=cve_threshold,
-            tablefmt='simple',
+            tablefmt='html',
+        )
+
+        # get product risk report
+        ci.util.info('retrieving product risk report')
+        prr = ws_client.get_product_risk_report(product_token=product_token)
+
+        # assemble html body
```

Comment:
nit: I think this comment might be omitted, as it merely repeats the called function's name (which is an indication that said function is well-named)

Reply:
you are right. I removed a couple of obvious comments